django/db/models/fields/json.py

```diff
@@ -0,0 +1,489 @@
+                params.append(val)
+            return ' AND '.join(conditions), params
+        else:
+            return '%s = %%s' % lhs, [self.rhs]
+
+
+@JSONField.register_lookup
+class ContainedBy(DataContains):
+    lookup_name = 'contained_by'
+    postgres_operator = '<@'
+
+    def as_mysql(self, compiler, connection):
+        return super().as_mysql(compiler, connection, flipped=True)
+
+
+class JSONValue(Func):
+    function = 'CAST'
+    template = '%(function)s(%(expressions)s AS JSON)'
+
+    def __init__(self, expression):
+        super().__init__(Value(expression))
+
+
+@JSONField.register_lookup
+class JSONExact(lookups.Exact):
+    can_use_none_as_rhs = True
+
+    def process_lhs(self, compiler, connection):
+        lhs, lhs_params = super().process_lhs(compiler, connection)
+        if connection.vendor == 'sqlite':
+            rhs, rhs_params = super().process_rhs(compiler, connection)
+            if rhs == '%s' and rhs_params == [None]:
+                # Need to use JSON_TYPE instead of JSON_EXTRACT
+                # to determine JSON null values.
+                lhs = "JSON_TYPE(%s, '$')" % lhs
+        return lhs, lhs_params
+
+    def process_rhs(self, compiler, connection):
+        rhs, rhs_params = super().process_rhs(compiler, connection)
+        # Treat None lookup values as null.
+        if rhs == '%s' and rhs_params == [None]:
+            rhs, rhs_params = ('%s', ['null'])
+        if connection.vendor == 'mysql':
+            func = []
+            for value in rhs_params:
+                obj = json.loads(value)
+                if connection.mysql_is_mariadb and isinstance(obj, str):
+                    func.append("JSON_UNQUOTE(JSON_EXTRACT(%s, '$'))")
+                else:
+                    func.append("JSON_EXTRACT(%s, '$')")
+            rhs = rhs % tuple(func)
+        return rhs, rhs_params
+
+
+class KeyTransform(Transform):
+    postgres_operator = '->'
+    postgres_nested_operator = '#>'
+
+    def __init__(self, key_name, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.key_name = key_name
+
+    def _preprocess_lhs(self, compiler, connection):
+        key_transforms = [self.key_name]
+        previous = self.lhs
+        while isinstance(previous, KeyTransform):
+            key_transforms.insert(0, previous.key_name)
+            previous = previous.lhs
+        lhs, params = compiler.compile(previous)
+        return lhs, params, key_transforms
+
+    def as_mysql(self, compiler, connection):
+        lhs, params, key_transforms = self._preprocess_lhs(compiler, connection)
+        json_path = self.compile_json_path(key_transforms)
+        return 'JSON_EXTRACT(%s, %%s)' % lhs, params + [json_path]
+
+    def compile_json_path(self, key_transforms):
+        path = ['$']
+        for key_transform in key_transforms:
+            try:
+                num = int(key_transform)
+                path.append('[{}]'.format(num))
```

Comment:
This should be moved to an `else` branch to make it explicit that the `ValueError` is only expected on `int(...)`.